src/WebExploitation/FileUploads/Insufficient blacklisting.md

```diff
@@ -0,0 +1,88 @@
+
+
+## Table of Contents
+
+- [File Uploads: Unrestricted File Uploads!](#file-uploads-unrestricted-file-uploads)
+  - [Introduction](#Introduction)
+  - [Table of Contents](#table-of-contents)
+  - [How to identify](#how-to-identify)
+  - [Code/tools/websites](#codetoolswebsites)
+    - [Tools](#tools)
+    - [Websites](#websites)
+  - [Sample problem](#sample-problem)
+    - [Problem Statement](#problem-statement)
+    - [Solution](#solution)
+  - [References](#references)
+  - [Conclusion](#conclusion)
+
+
+## How to identify
+
+One of the more obvious ways of preventing users from uploading malicious scripts is to blacklist potentially dangerous file extensions like .php. The practice of blacklisting is inherently flawed as it's difficult to explicitly block every possible file extension that could be used to execute code. Such blacklists can sometimes be bypassed by using lesser known, alternative file extensions that may still be executable, such as .php5, .shtml, and so on. 
+
+**Upload Section** :
+Any type of upload can possibly have a upload vulnerability. Make sure you remember this when you encounter a upload section in real life.
+
+
+
+## Code/tools/websites
+
+### Tools
+- **Burp Suite:** A web vulnerability scanner with tools for manual testing.
+
+### Websites
+> https://portswigger.net/web-security/file-upload
+> https://ctf.hackthebox.com/ 
+
+
+## Sample problem
+
+### Problem Statement
+ This lab contains a vulnerable image upload function. Certain file extensions are blacklisted, but this defense can be bypassed due to a fundamental flaw in the configuration of this blacklist.. 
+
+
+- [Link to the Problem](https://portswigger.net/web-security/file-upload/lab-file-upload-web-shell-upload-via-extension-blacklist-bypass)
+
+### Solution
+Here we have a upload section under My Accounts tab to upload the Avatar.
+We use burpsuite proxy here to interupt the Requests. 
+
+- **Initial Request for JPG**:
+The Content-Type response header may provide clues as to what kind of file the server thinks it has served. If this header hasn't been explicitly set by the application code, it normally contains the result of the file extension/MIME type mapping. 
+  ```
+  Content-Disposition: form-data; name="avatar"; filename="filename.jpg"
+  Content-Type: image/jpeg
+  ``` 
+
+- **Restriction with Blacklisting**:
+This time when we try to upload this exploit directly...
+  ```php
+   <?php echo file_get_contents('/home/carlos/secret'); ?>
+  ```
+   We are not able to do so...
+
+- **The difference**:
+ Here we change the Content-Type to text/plain in burpsuite.
+   ```
+  Content-Disposition: form-data; name="avatar"; filename=".htaccess"
+  Content-Type: text/plain
+  ``` 
+    and we add 
```

Comment:
As a reader, it isn't clear what exactly you are doing here. Can we paste the final request from burpsuite here?